Performancecounter: panic: reflect.StructOf: field 0 has invalid name

### Current Behavior

Command:
C:\Users\Alexander\Desktop\windows_exporter-0.30.0-rc.4-amd64.exe --config.file=C:\Users\Alexander\Desktop\windows_exporter.yml --log.file=C:\Users\Alexander\Desktop\windows_exporter.log

config:

I tried with

Panic
`    objects: |-`

Panic
`    objects: |`

Running but not perfcounter results.
`    objects:`


Discrepancy in readme:
https://github.com/prometheus-community/windows_exporter/blob/master/docs/collector.performancecounter.md

top of readme ( |-   vs   |  )
```
Caution

If you are using a configuration file, the value must be kept as a string.

Use a |- to keep the value as a string.
```

bottom of readme
```
Note

If you are using a configuration file, the value must be keep as string.

Example:

collector:
  performancecounter:
    objects: |
```



```
## C:\Users\Alexander\Desktop\windows_exporter-0.30.0-rc.4-amd64.exe --config.file=C:\Users\Alexander\Desktop\windows_exporter.yml --log.file=C:\Users\Alexander\Desktop\windows_exporter.log


collectors:
  enabled: performancecounter

## https://kb.paessler.com/en/topic/50673-how-can-i-find-out-the-names-of-available-performance-counters
## https://gist.github.com/shadowbq/7babb99078721c153346fed4173059ad
## per Windows cmd PerformanceCounter auslesen. Beispiel case-insensitive mit regex für BitLocker                 typeperf -qx | findstr /I /R BitLocker
## https://leanpub.com/windowspowershellnetworkingguide/read#leanpub-auto-performance-counters
## Powershell alle Performance Counter:
##                           Get-Counter -ListSet * | Sort-Object CounterSetName | Select-Object CounterSetName
## Powershell zu einem Performancecounter (hier: Prozess) die Pfade in Variable "paths":
##                           $paths = (Get-Counter -ListSet Prozess).paths
## Powershell alle Pfade aus de Variablen nun auslesen. Das ist das, was wir bei performancecounter brauchen
##                           Get-Counter -Counter $paths
collector:
  performancecounter:
### not working YAML
#    objects: |
#      - object: "Memory"
#        counters:
#          - name: "Cache Faults/sec"
#            type: "counter"
#      
#      - object: "BitLocker"
#        instances:
#          - "C:"
#          - "E:"
#          - "F:"
#        instance_label: "Laufwerk"
#        counters:
#          - name: "Write Requests/sec"
#            type: "counter"
#      
#      - object: "BitLocker"
#        instances:
#          - "D:"
#          - "G:"
#          - "H:"
#        instance_label: "DLaufwerk"
#        counters:
#          - name: "Read Requests/sec"
#            type: "counter"
#
### working JSON
    objects: |-
      [
        {
          "name": "pf1",
          "object": "Memory",
          "counters": [
            {
              "name": "Cache Faults/sec",
              "type": "counter"
            }
          ]
        },
        {
          "name": "pf2",
          "object": "BitLocker",
          "instances": [
            "*"
          ],
          "instance_label": "Laufwerk",
          "counters": [
            {
              "name": "Write Requests/sec",
              "type": "counter"
            },
            {
              "name": "Write Subrequests/sec",
              "type": "counter"
            },
            {
              "name": "Read Requests/sec",
              "type": "counter"
            },
            {
              "name": "Read Subrequests/sec",
              "type": "counter"
            }
          ]
        },
        {
          "name": "pf3",
          "object": "Event Tracing for Windows",
          "counters": [
            {
              "name": "Total Memory Usage --- Non-Paged Pool",
              "type": "counter"
            },
            {
              "name": "Total Memory Usage --- Paged Pool",
              "type": "counter"
            },
            {
              "name": "Total Number of Active Sessions",
              "type": "counter"
            },
            {
              "name": "Total Number of Distinct Disabled Providers",
              "type": "counter"
            }
          ]
        },
        {
          "name": "pf4",
          "object": "User Input Delay per Session",
          "counters": [
            {
              "name": "Max Session Input Delay (ms)",
              "type": "counter"
            }
          ]
        }
      ]

##
##
##      - object: "Processor Information"
##        instances: ["*"]
##        instance_label: "core"
##        counters:
##          - name: "% Processor Time"
##            type: "gauge"
##            metric: windows_performancecounter_processor_information_processor_time # optional
##            labels:
##              state: active
##          - name: "% Idle Time"
##            type: "counter"
##            metric: windows_performancecounter_processor_information_processor_time # optional
##            labels:
##              state: idle
##
##      - object: "BitLocker"
##        instances: ["C:"]
##        instance_label: "Laufwerk"
##        counters:
##          - name: "Schreibunteranforderungen/Sek."
##            metric: windows_performancecounter_bitlocker_leseanforderungen_seconds # optional
##            labels:
##              state: active
##
##      - object: "BitLocker"
##        instances: ["*"]
##        instance_label: "Laufwerk"
##        counters:
##          - name: "*"
##            #metric: windows_performancecounter_bitlocker_leseanforderungen_seconds # optional
##            labels:
##              state: active
##



## collectors:
##   enabled: ad,adcs,adfs,cache,cpu,cpu_info,cs,diskdrive,dfsr,dhcp,dns,exchange,filetime,fsrmquota,hyperv,iis,license,logical_disk,logon,memory,mscluster,msmq,mssql,netframework,net,os,pagefile,perfdata,physical_disk,printer,process,remote_fx,scheduled_task,service,smb,smbclient,smtp,system,tcp,terminal_services,thermalzone,time,udp,update,vmware
## collector:
##   process:
##     include: .*alloy.*
##   service:
##     exclude: ".*"
##   scheduled_task:
##     exclude: ".*"
##   mssql:
##     enabled_classes: ["accessmethods", "availreplica", "bufman", "databases", "dbreplica", "genstats", "locks", "memmgr", "sqlstats", "sqlerrors", "transactions", "waitstats"]
##   smb:
##     enabled_list: "ServerShares"
##   smb_client:
##     enabled_list: "ClientShares"
##   hyperv:
##     enabled: datastore,dynamic_memory_balancer,dynamic_memory_vm,hypervisor_logical_processor,hypervisor_root_partition,hypervisor_root_virtual_processor,hypervisor_virtual_processor,legacy_network_adapter,virtual_machine_health_summary,virtual_machine_vid_partition,virtual_network_adapter,virtual_network_adapter_drop_reasons,virtual_smb,virtual_storage_device,virtual_switch
##   filetime:
##     file-patterns: C:\Users\Alexander\Desktop\windows_exporter.log


log:
  level: debug
scrape:
  timeout-margin: 0.5
telemetry:
  path: /metrics
  max-requests: 0
web:
  listen-address: ":9182"
  disable-exporter-metrics: true     ## disabled go, prom metrics

```


Error:
```
C:\WINDOWS\system32>C:\Users\Alexander\Desktop\windows_exporter-0.30.0-rc.4-amd64.exe --config.file=C:\Users\Alexander\Desktop\windows_exporter.yml --log.file=C:\Users\Alexander\Desktop\windows_exporter.log
panic: reflect.StructOf: field 0 has invalid name

goroutine 9 [running]:
reflect.StructOf({0xc0000f8000, 0x2, 0xc000047f18?})
        C:/hostedtoolcache/windows/go/1.23.4/x64/src/reflect/type.go:2371 +0x2694
github.com/prometheus-community/windows_exporter/internal/collector/performancecounter.(*Collector).Build(0xc000144e60, 0xc000243e00, 0xdd9440?)
        D:/a/windows_exporter/windows_exporter/internal/collector/performancecounter/performancecounter.go:193 +0x1525
github.com/prometheus-community/windows_exporter/pkg/collector.(*Collection).Build.func1()
        D:/a/windows_exporter/windows_exporter/pkg/collector/collection.go:220 +0x8a
created by github.com/prometheus-community/windows_exporter/pkg/collector.(*Collection).Build in goroutine 1
        D:/a/windows_exporter/windows_exporter/pkg/collector/collection.go:217 +0x1e5

C:\WINDOWS\system32>
C:\WINDOWS\system32>
C:\WINDOWS\system32>C:\Users\Alexander\Desktop\windows_exporter-0.30.0-rc.4-amd64.exe --config.file=C:\Users\Alexander\Desktop\windows_exporter.yml --log.file=C:\Users\Alexander\Desktop\windows_exporter.log
panic: reflect.StructOf: field 0 has invalid name

goroutine 23 [running]:
reflect.StructOf({0xc000328000, 0x2, 0xc000047f18?})
        C:/hostedtoolcache/windows/go/1.23.4/x64/src/reflect/type.go:2371 +0x2694
github.com/prometheus-community/windows_exporter/internal/collector/performancecounter.(*Collector).Build(0xc0000f4f00, 0xc0001f7ea0, 0xdd9440?)
        D:/a/windows_exporter/windows_exporter/internal/collector/performancecounter/performancecounter.go:193 +0x1525
github.com/prometheus-community/windows_exporter/pkg/collector.(*Collection).Build.func1()
        D:/a/windows_exporter/windows_exporter/pkg/collector/collection.go:220 +0x8a
created by github.com/prometheus-community/windows_exporter/pkg/collector.(*Collection).Build in goroutine 1
        D:/a/windows_exporter/windows_exporter/pkg/collector/collection.go:217 +0x1e5

C:\WINDOWS\system32>
C:\WINDOWS\system32>
C:\WINDOWS\system32>C:\Users\Alexander\Desktop\windows_exporter-0.30.0-rc.4-amd64.exe --config.file=C:\Users\Alexander\Desktop\windows_exporter.yml --log.file=C:\Users\Alexander\Desktop\windows_exporter.log
panic: reflect.StructOf: field 0 has invalid name

goroutine 7 [running]:
reflect.StructOf({0xc000270000, 0x2, 0xc0003c7f18?})
        C:/hostedtoolcache/windows/go/1.23.4/x64/src/reflect/type.go:2371 +0x2694
github.com/prometheus-community/windows_exporter/internal/collector/performancecounter.(*Collector).Build(0xc000338640, 0xc0003ac3f0, 0xdd9440?)
        D:/a/windows_exporter/windows_exporter/internal/collector/performancecounter/performancecounter.go:193 +0x1525
github.com/prometheus-community/windows_exporter/pkg/collector.(*Collection).Build.func1()
        D:/a/windows_exporter/windows_exporter/pkg/collector/collection.go:220 +0x8a
created by github.com/prometheus-community/windows_exporter/pkg/collector.(*Collection).Build in goroutine 1
        D:/a/windows_exporter/windows_exporter/pkg/collector/collection.go:217 +0x1e5
```


### Expected Behavior

metrics, no panic, 

### Steps To Reproduce

```Markdown
see above
```


### Environment

- windows_exporter Version: windows_exporter-0.30.0-rc.4-amd64.exe
- Windows Version: `10`


### windows_exporter logs

```shell
time=2024-12-27T19:53:15.522Z level=INFO source=config.go:79 msg="loading configuration file: C:\\Users\\Alexander\\Desktop\\windows_exporter.yml"
time=2024-12-27T19:53:15.523Z level=DEBUG source=main.go:196 msg="logging has Started"
time=2024-12-27T19:53:15.523Z level=DEBUG source=main.go:337 msg="setting process priority to normal"
time=2024-12-27T19:53:15.527Z level=ERROR source=main.go:218 msg="couldn't initialize collector" err="error build collector performancecounter: object name is required"
time=2024-12-27T19:53:23.115Z level=INFO source=config.go:79 msg="loading configuration file: C:\\Users\\Alexander\\Desktop\\windows_exporter.yml"
time=2024-12-27T19:53:23.116Z level=DEBUG source=main.go:196 msg="logging has Started"
time=2024-12-27T19:53:23.116Z level=DEBUG source=main.go:337 msg="setting process priority to normal"
time=2024-12-27T19:53:23.120Z level=ERROR source=main.go:218 msg="couldn't initialize collector" err="error build collector performancecounter: object name is required"
time=2024-12-27T19:54:28.866Z level=INFO source=config.go:79 msg="loading configuration file: C:\\Users\\Alexander\\Desktop\\windows_exporter.yml"
time=2024-12-27T19:54:28.867Z level=DEBUG source=main.go:196 msg="logging has Started"
time=2024-12-27T19:54:28.867Z level=DEBUG source=main.go:337 msg="setting process priority to normal"
time=2024-12-27T19:54:29.142Z level=ERROR source=main.go:218 msg="couldn't initialize collector" err="error build collector performancecounter: object name is required"
time=2024-12-27T19:55:07.954Z level=INFO source=config.go:79 msg="loading configuration file: C:\\Users\\Alexander\\Desktop\\windows_exporter.yml"
time=2024-12-27T19:55:07.955Z level=DEBUG source=main.go:196 msg="logging has Started"
time=2024-12-27T19:55:07.955Z level=DEBUG source=main.go:337 msg="setting process priority to normal"
time=2024-12-27T19:55:08.230Z level=ERROR source=main.go:218 msg="couldn't initialize collector" err="error build collector performancecounter: object name is required"
time=2024-12-27T19:57:10.635Z level=INFO source=config.go:79 msg="loading configuration file: C:\\Users\\Alexander\\Desktop\\windows_exporter.yml"
time=2024-12-27T19:57:10.636Z level=DEBUG source=main.go:196 msg="logging has Started"
time=2024-12-27T19:57:10.636Z level=DEBUG source=main.go:337 msg="setting process priority to normal"
time=2024-12-27T19:57:10.904Z level=ERROR source=main.go:218 msg="couldn't initialize collector" err="error build collector performancecounter: object name is required"
time=2024-12-27T19:57:43.346Z level=INFO source=config.go:79 msg="loading configuration file: C:\\Users\\Alexander\\Desktop\\windows_exporter.yml"
time=2024-12-27T19:57:43.347Z level=DEBUG source=main.go:196 msg="logging has Started"
time=2024-12-27T19:57:43.347Z level=DEBUG source=main.go:337 msg="setting process priority to normal"
time=2024-12-27T19:57:43.355Z level=INFO source=main.go:311 msg="Running as NACHTFALKE\\Alexander"
time=2024-12-27T19:57:43.355Z level=INFO source=main.go:228 msg="Enabled collectors: performancecounter"
time=2024-12-27T19:57:43.355Z level=INFO source=main.go:246 msg="starting windows_exporter in 9.7822ms" version=0.30.0-rc.4 branch=HEAD revision=78386557d4dcb168b82e87084db8faf04dacdd72 goversion=go1.23.4 builddate=20241222-13:19:26 maxprocs=12
time=2024-12-27T19:57:43.357Z level=INFO source=tls_config.go:347 msg="Listening on" address=[::]:9182
time=2024-12-27T19:57:43.357Z level=INFO source=tls_config.go:350 msg="TLS is disabled." http2=false address=[::]:9182
time=2024-12-27T19:57:56.653Z level=DEBUG source=collect.go:220 msg="collector performancecounter succeeded after 0s, resulting in 0 metrics"
time=2024-12-27T19:58:30.486Z level=DEBUG source=collect.go:220 msg="collector performancecounter succeeded after 0s, resulting in 0 metrics"
time=2024-12-27T20:00:00.159Z level=INFO source=main.go:278 msg="Shutting down windows_exporter via kill signal"
time=2024-12-27T20:00:00.159Z level=INFO source=main.go:296 msg="windows_exporter has shut down"
time=2024-12-27T20:00:55.715Z level=INFO source=config.go:79 msg="loading configuration file: C:\\Users\\Alexander\\Desktop\\windows_exporter.yml"
time=2024-12-27T20:00:55.717Z level=DEBUG source=main.go:196 msg="logging has Started"
time=2024-12-27T20:00:55.717Z level=DEBUG source=main.go:337 msg="setting process priority to normal"
time=2024-12-27T20:01:40.411Z level=INFO source=config.go:79 msg="loading configuration file: C:\\Users\\Alexander\\Desktop\\windows_exporter.yml"
time=2024-12-27T20:01:40.412Z level=DEBUG source=main.go:196 msg="logging has Started"
time=2024-12-27T20:01:40.412Z level=DEBUG source=main.go:337 msg="setting process priority to normal"
time=2024-12-27T20:01:59.123Z level=INFO source=config.go:79 msg="loading configuration file: C:\\Users\\Alexander\\Desktop\\windows_exporter.yml"
time=2024-12-27T20:01:59.124Z level=DEBUG source=main.go:196 msg="logging has Started"
time=2024-12-27T20:01:59.124Z level=DEBUG source=main.go:337 msg="setting process priority to normal"
time=2024-12-27T20:02:33.716Z level=INFO source=config.go:79 msg="loading configuration file: C:\\Users\\Alexander\\Desktop\\windows_exporter.yml"
time=2024-12-27T20:02:33.717Z level=DEBUG source=main.go:196 msg="logging has Started"
time=2024-12-27T20:02:33.717Z level=DEBUG source=main.go:337 msg="setting process priority to normal"
time=2024-12-27T20:05:54.677Z level=INFO source=config.go:79 msg="loading configuration file: C:\\Users\\Alexander\\Desktop\\windows_exporter.yml"
time=2024-12-27T20:05:54.678Z level=DEBUG source=main.go:196 msg="logging has Started"
time=2024-12-27T20:05:54.678Z level=DEBUG source=main.go:337 msg="setting process priority to normal"
time=2024-12-27T20:05:54.689Z level=INFO source=main.go:311 msg="Running as NACHTFALKE\\Alexander"
time=2024-12-27T20:05:54.689Z level=INFO source=main.go:228 msg="Enabled collectors: performancecounter"
time=2024-12-27T20:05:54.689Z level=INFO source=main.go:246 msg="starting windows_exporter in 12.912ms" version=0.30.0-rc.4 branch=HEAD revision=78386557d4dcb168b82e87084db8faf04dacdd72 goversion=go1.23.4 builddate=20241222-13:19:26 maxprocs=12
time=2024-12-27T20:05:54.690Z level=INFO source=tls_config.go:347 msg="Listening on" address=[::]:9182
time=2024-12-27T20:05:54.690Z level=INFO source=tls_config.go:350 msg="TLS is disabled." http2=false address=[::]:9182
time=2024-12-27T20:06:00.272Z level=DEBUG source=collect.go:220 msg="collector performancecounter succeeded after 0s, resulting in 0 metrics"
time=2024-12-27T20:06:02.079Z level=DEBUG source=collect.go:220 msg="collector performancecounter succeeded after 0s, resulting in 0 metrics"
time=2024-12-27T20:06:04.043Z level=DEBUG source=collect.go:220 msg="collector performancecounter succeeded after 0s, resulting in 0 metrics"
time=2024-12-27T20:06:04.283Z level=DEBUG source=collect.go:220 msg="collector performancecounter succeeded after 0s, resulting in 0 metrics"
time=2024-12-27T20:06:04.442Z level=DEBUG source=collect.go:220 msg="collector performancecounter succeeded after 0s, resulting in 0 metrics"
time=2024-12-27T20:06:04.594Z level=DEBUG source=collect.go:220 msg="collector performancecounter succeeded after 0s, resulting in 0 metrics"
```


### Anything else?

_No response_

--- a/internal/collector/performancecounter/performancecounter.go
+++ b/internal/collector/performancecounter/performancecounter.go
@@ -20,6 +20,7 @@ import (
 	"fmt"
 	"log/slog"
 	"reflect"
+	"regexp"
 	"slices"
 	"strings"
 	"time"
@@ -33,6 +34,8 @@ import (
 )
 
 const Name = "performancecounter"
+
+var reNonAlphaNum = regexp.MustCompile(`[^a-zA-Z0-9]`)
 
 type Config struct {
 	Objects []Object `yaml:"objects"`
@@ -115,15 +118,6 @@ func (c *Collector) Close() error {
 
 func (c *Collector) Build(logger *slog.Logger, _ *mi.Session) error {
 	c.logger = logger.With(slog.String("collector", Name))
-
-	c.metricNameReplacer = strings.NewReplacer(
-		".", "",
-		"%", "",
-		"/", "_",
-		" ", "_",
-		"-", "_",
-	)
-
 	c.objects = make([]Object, 0, len(c.config.Objects))
 	names := make([]string, 0, len(c.config.Objects))
 
@@ -152,7 +146,7 @@ func (c *Collector) Build(logger *slog.Logger, _ *mi.Session) error {
 
 		for j, counter := range object.Counters {
 			if counter.Metric == "" {
-				c.config.Objects[i].Counters[j].Metric = c.sanitizeMetricName(
+				c.config.Objects[i].Counters[j].Metric = sanitizeMetricName(
 					fmt.Sprintf("%s_%s_%s_%s", types.Namespace, Name, object.Object, counter.Name),
 				)
 			}
@@ -171,11 +165,28 @@ func (c *Collector) Build(logger *slog.Logger, _ *mi.Session) error {
 			}
 
 			counters = append(counters, counter.Name)
-			fields = append(fields, reflect.StructField{
-				Name: strings.ToUpper(c.sanitizeMetricName(counter.Name)),
-				Type: reflect.TypeOf(float64(0)),
-				Tag:  reflect.StructTag(fmt.Sprintf(`perfdata:"%s"`, counter.Name)),
-			})
+
+			field, err := func(name string) (field reflect.StructField, err error) {
+				defer func() {
+					if r := recover(); r != nil {
+						err = fmt.Errorf("failed to create field for %s: %v", name, r)
+					}
+				}()
+
+				return reflect.StructField{
+					Name: strings.ToUpper(sanitizeMetricName(name)),
+					Type: reflect.TypeOf(float64(0)),
+					Tag:  reflect.StructTag(fmt.Sprintf(`perfdata:"%s"`, name)),
+				}, nil
+			}(counter.Name)
+
+			if err != nil {
+				errs = append(errs, err)
+
+				continue
+			}
+
+			fields = append(fields, field)
 		}
 
 		if object.Instances != nil {
@@ -276,7 +287,7 @@ func (c *Collector) collectObject(ch chan<- prometheus.Metric, perfDataObject Ob
 		for _, counter := range perfDataObject.Counters {
 			val := reflect.ValueOf(sliceValue).Index(i)
 
-			field := val.FieldByName(strings.ToUpper(c.sanitizeMetricName(counter.Name)))
+			field := val.FieldByName(strings.ToUpper(sanitizeMetricName(counter.Name)))
 			if !field.IsValid() {
 				errs = append(errs, fmt.Errorf("%s not found in collected data", counter.Name))
 
@@ -355,6 +366,6 @@ func (c *Collector) collectObject(ch chan<- prometheus.Metric, perfDataObject Ob
 	return errors.Join(errs...)
 }
 
-func (c *Collector) sanitizeMetricName(name string) string {
-	return strings.Trim(c.metricNameReplacer.Replace(strings.ToLower(name)), "_")
+func sanitizeMetricName(name string) string {
+	return strings.Trim(reNonAlphaNum.ReplaceAllString(strings.ToLower(name), "_"), "_")
 }

--- a/internal/collector/performancecounter/performancecounter.go
+++ b/internal/collector/performancecounter/performancecounter.go
@@ -54,8 +54,6 @@ type Collector struct {
 
 	objects []Object
 
-	metricNameReplacer *strings.Replacer
-
 	// meta
 	subCollectorScrapeDurationDesc *prometheus.Desc
 	subCollectorScrapeSuccessDesc  *prometheus.Desc
@@ -166,7 +164,7 @@ func (c *Collector) Build(logger *slog.Logger, _ *mi.Session) error {
 
 			counters = append(counters, counter.Name)
 
-			field, err := func(name string) (field reflect.StructField, err error) {
+			field, err := func(name string) (_ reflect.StructField, err error) {
 				defer func() {
 					if r := recover(); r != nil {
 						err = fmt.Errorf("failed to create field for %s: %v", name, r)
@@ -179,7 +177,6 @@ func (c *Collector) Build(logger *slog.Logger, _ *mi.Session) error {
 					Tag:  reflect.StructTag(fmt.Sprintf(`perfdata:"%s"`, name)),
 				}, nil
 			}(counter.Name)
-
 			if err != nil {
 				errs = append(errs, err)
 

--- a/internal/collector/performancecounter/performancecounter_test_test.go
+++ b/internal/collector/performancecounter/performancecounter_test_test.go
@@ -129,10 +129,10 @@ windows_performancecounter_processor_information_processor_time\{core="0,0",stat
 		},
 		{
 			name:            "counter with spaces and brackets",
-			object:          "User Input Delay per Session",
+			object:          "invalid",
 			instances:       nil,
-			buildErr:        pdh.NewPdhError(pdh.CstatusNoCounter).Error(),
-			counters:        []performancecounter.Counter{{Name: "Max Session Input Delay (ms)", Type: "counter"}},
+			buildErr:        pdh.NewPdhError(pdh.CstatusNoObject).Error(),
+			counters:        []performancecounter.Counter{{Name: "Total Memory Usage --- Non-Paged Pool", Type: "counter"}, {Name: "Max Session Input Delay (ms)", Type: "counter"}},
 			expectedMetrics: nil,
 		},
 	} {

--- a/internal/collector/performancecounter/performancecounter_test_test.go
+++ b/internal/collector/performancecounter/performancecounter_test_test.go
@@ -25,6 +25,7 @@ import (
 	"testing"
 
 	"github.com/prometheus-community/windows_exporter/internal/collector/performancecounter"
+	"github.com/prometheus-community/windows_exporter/internal/pdh"
 	"github.com/prometheus/client_golang/prometheus"
 	"github.com/prometheus/client_golang/prometheus/promhttp"
 	"github.com/stretchr/testify/assert"
@@ -124,6 +125,14 @@ windows_performancecounter_processor_information_processor_time\{core="0,0",stat
 			instances:       nil,
 			buildErr:        "counter name Available Bytes is duplicated",
 			counters:        []performancecounter.Counter{{Name: "Available Bytes", Type: "gauge"}, {Name: "Available Bytes", Type: "gauge"}},
+			expectedMetrics: nil,
+		},
+		{
+			name:            "counter with spaces and brackets",
+			object:          "User Input Delay per Session",
+			instances:       nil,
+			buildErr:        pdh.NewPdhError(pdh.CstatusNoCounter).Error(),
+			counters:        []performancecounter.Counter{{Name: "Max Session Input Delay (ms)", Type: "counter"}},
 			expectedMetrics: nil,
 		},
 	} {

--- a/internal/collector/vmware/vmware.go
+++ b/internal/collector/vmware/vmware.go
@@ -93,11 +93,19 @@ func (c *Collector) Close() error {
 }
 
 func (c *Collector) Build(_ *slog.Logger, _ *mi.Session) error {
-	var err error
+	var (
+		err  error
+		errs []error
+	)
 
 	c.perfDataCollectorCPU, err = pdh.NewCollector[perfDataCounterValuesCPU]("VM Processor", pdh.InstancesTotal)
 	if err != nil {
-		return fmt.Errorf("failed to create VM Processor collector: %w", err)
+		errs = append(errs, fmt.Errorf("failed to create VM Processor collector: %w", err))
+	}
+
+	c.perfDataCollectorMemory, err = pdh.NewCollector[perfDataCounterValuesMemory]("VM Memory", nil)
+	if err != nil {
+		errs = append(errs, fmt.Errorf("failed to create VM Memory collector: %w", err))
 	}
 
 	c.cpuLimitMHz = prometheus.NewDesc(
@@ -142,11 +150,6 @@ func (c *Collector) Build(_ *slog.Logger, _ *mi.Session) error {
 		nil,
 		nil,
 	)
-
-	c.perfDataCollectorMemory, err = pdh.NewCollector[perfDataCounterValuesMemory]("VM Memory", nil)
-	if err != nil {
-		return fmt.Errorf("failed to create VM Memory collector: %w", err)
-	}
 
 	c.memActive = prometheus.NewDesc(
 		prometheus.BuildFQName(types.Namespace, Name, "mem_active_bytes"),
@@ -221,7 +224,7 @@ func (c *Collector) Build(_ *slog.Logger, _ *mi.Session) error {
 		nil,
 	)
 
-	return nil
+	return errors.Join(errs...)
 }
 
 // Collect sends the metric values for each metric

